### ISSUE TITLE
Notice: Undefined index: refund_id in catalog/model/payment/mollie/base.php on line 210

### ISSUE THREAD

**Issue by inekestijnman** [mollie/OpenCart#164]
I am still in testing mode. (opencart 2.3.0.2 Mollie 9.1.1)

After clicking cancel payment it successfully returned to the "Your payment has not been completed" form. But there was on top an error for undefined index in mollie base.php

Notice: Undefined index: refund_id in /home/u44466p39874/domains/gemc.eu/public_html/test2/catalog/model/payment/mollie/base.php on line 210 Notice: A non well formed numeric value encountered in /home/u44466p39874/domains/gemc.eu/public_html/test2/catalog/model/extension/total/shipping.php on line 24

![afbeelding](https://user-images.githubusercontent.com/46712328/64783032-ff699200-d566-11e9-8765-1d51753176d2.png)

**Comment by Krokit** [comment]
@inekestijnman I made some type of 'hacking' for a friend who uses open cart, but i don't know if this refund_id is returned in any way. so i made it with if-else.

I'm checking if refund_id is available in the array if not, the query will be executed without the part with refund_id.


```php 
            if(!in_array('refund_id', $data)) {
                $this->db->query(
                    sprintf(
                        "UPDATE `%smollie_payments` 
					 SET `transaction_id` = '%s', `bank_status` = '%s'
					 WHERE `order_id` = '%s' AND `mollie_order_id` = '%s';",
                        DB_PREFIX,
                        $this->db->escape($data['payment_id']),
                        $this->db->escape($data['status']),

                        $this->db->escape($order_id),
                        $this->db->escape($mollie_order_id)
                    )
                );
            } else {
                $this->db->query(
                    sprintf(
                        "UPDATE `%smollie_payments` 
					 SET `transaction_id` = '%s', `bank_status` = '%s', `refund_id` = '%s'
					 WHERE `order_id` = '%s' AND `mollie_order_id` = '%s';",
                        DB_PREFIX,
                        $this->db->escape($data['payment_id']),
                        $this->db->escape($data['status']),
                        $this->db->escape($data['refund_id']),

                        $this->db->escape($order_id),
                        $this->db->escape($mollie_order_id)
                    )
                );
            }
```

**Comment by inekestijnman** [comment]
Fixed in 9.1.2
